contracts/StablecoinConverter.sol

```diff
@@ -260,31 +279,39 @@ contract StablecoinConverter is EpochTokenLocker {
                     currentPrices[order.buyToken],
                     currentPrices[order.sellToken]
                 );
+                buyVolume = uint128(buyVolume.mul(feeDenominator - 1) / feeDenominator);
                 revertRemainingOrder(owner, orderId, sellVolume);
                 subtractBalance(owner, tokenIdToAddressMap(order.buyToken), buyVolume);
             }
         }
     }
 
     function findPriceIndex(uint16 index, uint16[] memory tokenIdForPrice) private pure returns (uint) {
+        // return fee token
+        if (index == 0) {
+            return 0;
+        }
+        // binary search for the other tokens
```

Comment:
Very nice! When did this happen! reduction from O(n) to O(log n)! Sweet

Reply:
Felix was pushing for it :)

Reply:
https://github.com/gnosis/dex-contracts/pull/164#pullrequestreview-273126587